src/main/java/org/jabref/logic/util/io/RegExpBasedFileFinder.java

```diff
@@ -195,30 +218,14 @@ public static String expandBrackets(String bracketString, BibEntry entry, BibDat
             } // End process directory information
         }
 
-        // Last step: check if the given file can be found in this directory
-        String filePart = fileParts[fileParts.length - 1].replace("[extension]", EXT_MARKER);
-        String expandedBracketAsRegexpLiterals = BracketedPattern.expandBrackets(filePart,
-                expandBracketContentToRegexpLiterals(entry));
-        String filenameToLookFor = expandedBracketAsRegexpLiterals.replaceAll(EXT_MARKER, extensionRegExp);
-
-        try {
-            final Pattern toMatch = Pattern.compile('^' + filenameToLookFor.replaceAll("\\\\\\\\", "\\\\") + '$',
-                    Pattern.CASE_INSENSITIVE);
-            BiPredicate<Path, BasicFileAttributes> matcher = (path, attributes) -> toMatch.matcher(path.getFileName().toString()).matches();
-            resultFiles.addAll(collectFilesWithMatcher(actualDirectory, matcher));
-        } catch (UncheckedIOException | PatternSyntaxException e) {
-            throw new IOException("Could not look for " + filenameToLookFor, e);
-        }
-
-        return resultFiles;
-    }
-
-    private List<Path> collectFilesWithMatcher(Path actualDirectory, BiPredicate<Path, BasicFileAttributes> matcher) {
+        Pattern toMatch = createFileNamePattern(fileParts, extensionRegExp, entry);
+        BiPredicate<Path, BasicFileAttributes> matcher = (path, attributes) -> toMatch.matcher(path.getFileName().toString()).matches();
         try (Stream<Path> pathStream = Files.find(actualDirectory, 1, matcher, FileVisitOption.FOLLOW_LINKS)) {
-            return pathStream.collect(Collectors.toList());
-        } catch (UncheckedIOException | IOException ioe) {
-            return Collections.emptyList();
```

Comment:
I did not find a reason why this should return an empty list in #3623. If it does return an empty list, in effect, all checked/unchecked IOErrors are swallowed and only PatternSyntaxException (that are re-thrown as IOExceptions) can occur.

Reply:
Could you add that as comment to `throw`?  That way, the idea is not lost.

Reply:
👍 
I believe the now added comment is clear regarding what cases an empty list might be returned instead of an exception being thrown.